content_sync/apis/github.py

```diff
@@ -134,22 +154,44 @@ def upsert_content_files_for_user(  # pylint:disable=too-many-locals
         """
         Upsert multiple WebsiteContent objects to github in one commit
         """
+        unsynced_states = ContentSyncState.objects.filter(
+            Q(content__website=self.website) & Q(content__updated_by=user_id)
+        ).exclude(
+            Q(current_checksum=F("synced_checksum")) & Q(synced_checksum__isnull=False)
+        )
         repo = self.get_repo()
-        content_set = self.website.websitecontent_set.filter(updated_by__id=user_id)
         main_ref = repo.get_git_ref(f"heads/{settings.GIT_BRANCH_MAIN}")
         main_sha = main_ref.object.sha
         base_tree = repo.get_git_tree(main_sha)
-        element_list = list()
-        for content in content_set.iterator():
-            sync_state = content.content_sync_state
+        modified_element_list = []
+        content_filepaths = {}
+
+        for sync_state in unsynced_states.iterator():
+            content = sync_state.content
+            # Add any modified files
             if content.content_filepath and not sync_state.is_synced:
+                content_filepaths.update({content.id: content.content_filepath})
                 data = self.format_content_to_file(content)
-                element = InputGitTreeElement(
-                    content.content_filepath, "100644", "blob", data
+                modified_element_list.append(
+                    InputGitTreeElement(
+                        content.content_filepath, "100644", "blob", data
+                    )
                 )
-                element_list.append(element)
-        if len(element_list) > 0:
-            tree = repo.create_git_tree(element_list, base_tree)
+                # Remove the old filepath stored in the sync state data if it doesn't match current path
+                if (
+                    sync_state.data
+                    and sync_state.data.get(GIT_DATA_FILEPATH, None)
+                    != content.content_filepath
+                ):
+                    modified_element_list.append(
+                        InputGitTreeElement(
+                            sync_state.data["filepath"], "100644", "blob", sha=None
+                        )
+                    )
+                # TODO: Remove deleted content
```

Comment:
@rhysyngsun is this part of what you are working on or should I deal with it in this PR?

Reply:
Yeah, I'm working on that.